src/radio-group.test.interactions.ts

```diff
@@ -563,3 +564,163 @@ it('makes a radio tabbable when it is enabled programmatically and no other radi
+
+  assert(radios[1]);
+  radios[1].disabled = false;
+
+  expect(host.value).to.equal('two');
+});
+
+it('updates its `value` when a checked radio is programmatically disabled', async () => {
+  const host = await fixture<GlideCoreRadioGroup>(html`
+    <glide-core-radio-group label="Label">
+      <glide-core-radio-group-radio
+        label="One"
+        value="one"
+        checked
+      ></glide-core-radio-group-radio>
+
+      <glide-core-radio-group-radio
+        label="Two"
+        value="two"
+        checked
+      ></glide-core-radio-group-radio>
+    </glide-core-radio-group>
+  `);
+
+  const radios = host.querySelectorAll('glide-core-radio-group-radio');
+
+  assert(radios[1]);
+  radios[1].disabled = true;
+
+  expect(host.value).to.equal('one');
+});
+
+it('updates its `value` when the group is programmatically enabled', async () => {
+  const host = await fixture<GlideCoreRadioGroup>(html`
+    <glide-core-radio-group label="Label" disabled>
+      <glide-core-radio-group-radio
+        label="One"
+        value="one"
+        checked
+      ></glide-core-radio-group-radio>
+
+      <glide-core-radio-group-radio
+        label="Two"
+        value="two"
+        checked
+      ></glide-core-radio-group-radio>
+    </glide-core-radio-group>
+  `);
+
+  host.disabled = false;
+  await host.updateComplete;
+
+  expect(host.value).to.equal('two');
+});
+
+it('updates its `value` when the group is programmatically disabled', async () => {
+  const host = await fixture<GlideCoreRadioGroup>(html`
+    <glide-core-radio-group label="Label">
+      <glide-core-radio-group-radio
+        label="One"
+        value="one"
+        checked
+      ></glide-core-radio-group-radio>
+
+      <glide-core-radio-group-radio
+        label="Two"
+        value="two"
+      ></glide-core-radio-group-radio>
+    </glide-core-radio-group>
+  `);
+
+  host.disabled = true;
+  await host.updateComplete;
+
+  expect(host.value).to.equal('');
+});
+
+it('retains its `value` when the `value` of a checked radio that is not the last enabled one is changed', async () => {
+  const host = await fixture<GlideCoreRadioGroup>(html`
+    <glide-core-radio-group label="Label">
+      <glide-core-radio-group-radio
+        label="One"
+        value="one"
+        checked
+      ></glide-core-radio-group-radio>
+
+      <glide-core-radio-group-radio
+        label="Two"
+        value="two"
+        checked
+      ></glide-core-radio-group-radio>
+
+      <glide-core-radio-group-radio
+        label="Three"
+        value="three"
+        checked
+        disabled
+      ></glide-core-radio-group-radio>
+    </glide-core-radio-group>
+  `);
+
+  const radios = host.querySelectorAll('glide-core-radio-group-radio');
+
+  assert(radios[0]);
+  radios[0].value = 'test';
+
+  expect(host.value).to.equal('two');
+});
+
+it('updates its `value` when the `value` of the last checked and enabled radio is unchecked', async () => {
```

Comment:
The second `value` part reads a little odd to me because we're updating `checked` rather than value in this test case.  Something like this maybe? Or am I missing something?

```suggestion
it('updates its `value` when the last checked and enabled radio is unchecked', async () => {
```